src/Services/CreditCards/ICreditCardsServices.cs

```diff
@@ -0,0 +1,151 @@
+﻿using CloudDrive.Domain;
+using CloudDrive.Domain.Entities;
+using CloudDrive.Persistence;
+using Microsoft.EntityFrameworkCore;
+using Services.CreditCards;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CloudDrive.Services.CreditCards
+{
+    public interface ICreditCardsServices
+    {
+        Task<Result<CreditCardsDto>> GetUserCreditCardAsync(int id);
+        Task<Result<List<CreditCardsDto>>> GetUserCreditCardsForUserAsync(int id);
+        Task<Result<CreditCardsDto>> CreateUserCreditCard(UserCreditCard userCreditCard);
+        Task<Result> DeleteUserCreditCardAsync(int id);
+        Task<Result<CreditCardsDto>> UpdateUserCreditCardAsync(int id, UserCreditCard userCreditCard);
+    }
+
+    public class UserCreditCardsImpl : ICreditCardsServices
+    {
+        //Connect the Database 
+        private readonly AppDbContext _appDbContext;
+        public UserCreditCardsImpl(AppDbContext appDbContext)
+        {
+            _appDbContext = appDbContext;
+        }
+
+
+        public async Task<Result<CreditCardsDto>> CreateUserCreditCard(UserCreditCard userCreditCard)
+        {
+            _appDbContext.UserCreditCards.Add(userCreditCard);
```

Comment:
Please use transaction to commit data into the database to prevent bad data form being inserted in case of errors